### ISSUE TITLE
upgrade decaf377 deps

### ISSUE THREAD

**Issue by TalDerei** [penumbra-zone/web#1481]
so the original motivation for reworking the decaf377 crate was to eliminate our tightly coupled dependence on the Arkworks cryptography stack that’s used in our circuits. We redesigned the internal decaf377 API and generated our own formally verified code variants that we expected to be performant. Unfortunately, this led to some undesirable performance regressions for the proving time in the web, so we reverted some of that code. However, our changes were useful in enabling hardware wallet support on 32-bit targets (ledger).

tldr; the latest version of decaf377 doesn’t improve or degrade performance in the web, but the decaf377 crate dep should be bumped from `0.5.0` to `0.10.1` to maintain version parity with core.
